document-portal/file-transfer.c

```diff
@@ -144,15 +144,15 @@ static GHashTable *transfers;
 static FileTransfer *
 lookup_transfer (const char *key)
 {
-  FileTransfer *transfer;
+  g_autoptr(FileTransfer) transfer = NULL;
 
   G_LOCK (transfers);
   transfer = (FileTransfer *)g_hash_table_lookup (transfers, key);
   if (transfer)
     g_object_ref (transfer);
   G_UNLOCK (transfers);
 
-  return transfer;
+  return g_steal_pointer (&transfer);
```

Comment:
transfer actually isn't owning the FileTransfer for some time which is a bit awkward. Maybe it gets more readable by ref-ing at the return?
```C
  FileTransfer *transfer;

  G_LOCK (transfers);
  transfer = (FileTransfer *)g_hash_table_lookup (transfers, key);
  G_UNLOCK (transfers);
  
  if (!transfer)
    return NULL;

  return g_object_ref (transfer);
```

Reply:
Took the liberty to apply this change.